tests/DoctrineObjectTest.php

```diff
@@ -1781,6 +1751,22 @@ public function testHydrateOneToManyAssociationByReferenceUsingIdentifiersArrayF
     {
         // When using hydration by value, it will use the public API of the entity to set values (setters)
         $entity = new Assets\OneToManyEntity();
+
+        $reflSteps = [
+            new ReflectionClass(Assets\OneToManyEntity::class),
+            new ReflectionClass(Assets\ByValueDifferentiatorEntity::class),
+            new ReflectionClass(Assets\ByValueDifferentiatorEntity::class),
+            new ReflectionClass(Assets\OneToManyEntity::class),
+        ];
+        $this
+            ->metadata
+            ->method('getReflectionClass')
+            ->willReturnCallback(
+                static function () use (&$reflSteps) {
+                    return array_shift($reflSteps);
+                }
```

Comment:
Do you want to have a strict ordered result in this test in consecutive calls?

Reply:
For now, yes.

Reply:
Maybe `willReturnOnConsecutiveCalls` can be used here.

Reply:
Fixed with 734cd6b